src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import { FunctionComponent } from 'preact';
+import { useEffect, useRef, useState } from 'preact/hooks';
+import type { Toast as ToastPropType } from './types';
+import {
+  CheckCircleFilledIcon,
+  ClearCircleFilledIcon,
+} from '../../assets/icons';
+import { dismissToast } from '.';
+import { ToastType } from './enums';
+
+const prefersReducedMotion = () => {
+  const mediaQuery = matchMedia('(prefers-reduced-motion: reduce)');
+  return mediaQuery.matches;
+};
+
+const colorForToastType = (type: ToastType) => {
+  switch (type) {
+    case ToastType.Success:
+      return 'color-success';
+    case ToastType.Error:
+      return 'color-danger';
+    default:
+      return 'color-info';
+  }
+};
+
+const iconForToastType = (type: ToastType) => {
+  switch (type) {
+    case ToastType.Success:
+      return <CheckCircleFilledIcon className={colorForToastType(type)} />;
+    case ToastType.Error:
+      return <ClearCircleFilledIcon className={colorForToastType(type)} />;
+    case ToastType.Loading:
+      return <div className="sk-spinner w-4 h-4 spinner-info" />;
+    default:
+      return null;
+  }
+};
+
+const getToastPosition = (
+  boundingRect: DOMRect,
+  index: number
+): React.CSSProperties => {
+  const positionFromBottom = index > 0 ? boundingRect.height * index : 0;
+  const margin = index > 0 ? index * 0.45 : 0;
+
+  return {
+    bottom: `calc(${positionFromBottom}px + ${margin}rem)`,
+  };
+};
+
+type Props = {
+  toast: ToastPropType;
+  index: number;
+};
+
+export const Toast: FunctionComponent<Props> = ({ toast, index }) => {
+  const toastElementRef = useRef<HTMLDivElement>();
+
+  const icon = iconForToastType(toast.type);
+  const hasActions = toast?.actions?.length > 0;
```

Comment:
Can `toast` be undefined? Because on line 82 it's treated as a sure value. Also the type in Props doesn't have it as optional.